config.go

```diff
@@ -45,18 +45,13 @@ type Config struct {
 // ShouldUpgrade matches the scope and image for a container to decide if an upgrade is required
 func (c *Config) ShouldUpgrade(containerImage, containerScope string) bool {
 	if c.Image != containerImage {
-		switch {
-		case containerScope == "":
-			return true
-		case containerScope == AnyScope:
-			return true
-		case c.Scope == AnyScope:
-			return true
-		case containerScope == c.Scope:
-			return true
-		default:
+		if containerScope == AnyScope && c.Scope != containerScope {
 			return false
 		}
+		if containerScope == "" {
+			return true
+		}
+		return c.Scope == AnyScope
```

Comment:
if we ignore the CLI `update` flow this works fine, and that flow can be mitigated by making changes to the CLI - in either case, I think this incrementally gets use closer to a good end state (or maybe the final state) so this part lgtm.